internal/bminventory/inventory.go

```diff
@@ -4792,6 +4792,10 @@ func (b *bareMetalInventory) BindHostInternal(ctx context.Context, params instal
 	if err != nil {
 		return nil, common.NewApiError(http.StatusBadRequest, errors.Errorf("Failed to find cluster %s", params.BindHostParams.ClusterID))
 	}
+	allowed, err := b.authzHandler.HasAccessTo(ctx, cluster, auth.UpdateAction)
+	if err != nil || !allowed {
+		return nil, common.NewApiError(http.StatusBadRequest, errors.Errorf("Failed to find cluster %s", params.BindHostParams.ClusterID))
```

Comment:
Why are we returning BadRequest which is a code designed for malformed inputs? I think that we should return 403 or 404. This will help the client to understand what the problem is

Reply:
Changed to 404 as we don't want to expose any security ino in the api response.